### PR TITLE
HZN-1279: Make the telemetry groovy scripts reloadable

### PR DESCRIPTION
**Pull request by cgorantla**
Add generic fileUpdateWatcher,   use this watcher for groovy scripts.
When groovy script file updates, check and load updated script if it builds else use existing script.

* JIRA: http://issues.opennms.org/browse/HZN-1279

Our [continuous integration system](http://bamboo.internal.opennms.com:8085) will test and verify your changes.

### CODE REVIEW

**Comment by j-white** [comment]
Looks pretty good. It's nice to have this functionality in the `AbstractPersistingAdapter`, so that all of the adapters can easily take advantage of it.

File: core/lib/src/main/java/org/opennms/core/fileutils/FileUpdateWatcher.java [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+                for (WatchEvent<?> event : key.pollEvents()) {
+                    WatchEvent.Kind<?> kind = event.kind();
+                    WatchEvent<Path> ev = cast(event);
+                    Path dir = (Path) key.watchable();
+                    Path updatedFile = dir.resolve(ev.context());
+                    // Some editors create temporary buffers, watch for both create/modify
+                    if (kind == StandardWatchEventKinds.ENTRY_MODIFY || kind == StandardWatchEventKinds.ENTRY_CREATE) {
+                        if (updatedFile.toString().equals(fileName)) {
+                            final long now = System.currentTimeMillis();
+                            final long interval = Long.getLong("org.opennms.core.fileutils.watcher.interval.seconds", 1);
```
[the remark this answers is not in the document]
Comment (cgorantla) [reply core/lib/src/main/java/org/opennms/core/fileutils/FileUpdateWatcher.java]: On further testing, I didn't see much need for  'interval' , so removed this

**Comment by cgorantla** [comment]
@j-white  Handled all the review comments.

**Comment by j-white** [comment]
Thanks for the updates, works like a charm.

**Pull request merged**